src/SharpCompress/Archives/SevenZip/SevenZipArchive.cs

```diff
@@ -105,7 +105,7 @@ public static SevenZipArchive Open(Stream stream, ReaderOptions readerOptions =
     /// <param name="srcStream"></param>
     /// <param name="options"></param>
     internal SevenZipArchive(SourceStream srcStream)
-        : base(ArchiveType.SevenZip, srcStream) { }
+        : base(ArchiveType.Tar, srcStream) { }
```

Comment:
This isn't right.  We can't change archive types.

What do tars for gz and bz2 do?

Reply:
The GZipArchive does the same thing

Reply:
```
    /// <summary>
    /// Constructor with a SourceStream able to handle FileInfo and Streams.
    /// </summary>
    /// <param name="srcStream"></param>
    /// <param name="options"></param>
    internal GZipArchive(SourceStream srcStream)
        : base(ArchiveType.Tar, srcStream) { }

```